src/unit_tests/wazuh_db/test_wdb_global.c

```diff
@@ -4623,9 +4669,29 @@ void test_wdb_global_select_group_belong_exec_fail(void **state)
     expect_any_always(__wrap_sqlite3_bind_int, index);
     expect_any_always(__wrap_sqlite3_bind_int, value);
     will_return_always(__wrap_sqlite3_bind_int, SQLITE_OK);
-    will_return(__wrap_sqlite3_errmsg, "ERROR MESSAGE");
-    will_return(__wrap_wdb_exec_stmt_single_column, NULL);
-    expect_string(__wrap__mdebug1, formatted_msg, "wdb_exec_stmt(): ERROR MESSAGE");
+    /* wdb_exec_stmt_sized */
+    wrap_wdb_exec_stmt_sized_failed_call(STMT_SINGLE_COLUMN);
+
+    expect_string(__wrap__mdebug1, formatted_msg, "Failed in getting the groups of the agent.");
+
+    j_result = wdb_global_select_group_belong(data->wdb, 1);
+    assert_null(j_result);
+}
+
+void test_wdb_global_select_group_belong_socket_size_error(void **state)
+{
+    cJSON *j_result = NULL;
+    test_struct_t *data  = (test_struct_t *)*state;
+
+    will_return(__wrap_wdb_begin2, 1);
+    will_return(__wrap_wdb_stmt_cache, 1);
+    expect_any_always(__wrap_sqlite3_bind_int, index);
+    expect_any_always(__wrap_sqlite3_bind_int, value);
+    will_return_always(__wrap_sqlite3_bind_int, SQLITE_OK);
+    /* wdb_exec_stmt_sized */
+    wrap_wdb_exec_stmt_sized_socket_full_call(NULL, STMT_SINGLE_COLUMN);
+
+    expect_string(__wrap__mdebug2, formatted_msg, "The groups of the agent are bigger than the socket size.");
```

Comment:
Something like: "The agent groups exceed the socket maximum size response"

Reply:
This expression is much better, thank you. I'll just modify it to `The agent's groups exceed the socket maximum response size `.
The size and the amount of groups are limited so this case is really unlikely, but the message is there just to be consistent because there isn't a `last_id` parameter to ask again.